File: backend/src/main/java/com/tcc/alzheimer/model/exams/Exam.java
package com.tcc.alzheimer.model.exams;

import com.tcc.alzheimer.model.roles.Doctor;
import com.tcc.alzheimer.model.roles.Patient;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "exam")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Exam {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "exam_type_id", nullable = false)
    private ExamType type;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "exam_status_id", nullable = false)
    private ExamStatus status;

    @Column(name = "request_date", nullable = false)
    private LocalDateTime requestDate;

    @Column(name = "instructions", columnDefinition = "TEXT")
    private String instructions;

    @Column(name = "note", columnDefinition = "TEXT")
    private String note;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "updated_by")
    private Long updatedBy;

}

File: mobile/types/domain/exam.ts
import { ExamStatus } from "../enum/exam-status";
import { FileInfoDTO } from "./files";

export enum ExamType {
  BLOOD_TEST = "BLOOD_TEST",
  URINE_TEST = "URINE_TEST",
  BRAIN_SCAN = "BRAIN_SCAN",
  COGNITIVE_ASSESSMENT = "COGNITIVE_ASSESSMENT",
  OTHER = "OTHER",
}

export interface Exam {
  id?: string;
  doctorId: string;
  doctorName?: string;
  patientId: string;
  patientName?: string;
  examTypeId: ExamType;
  examTypeDescription?: string;
  examStatusId: ExamStatus;
  examStatusDescription?: string;
  requestDate: string;
  instructions?: string;
  result?: string;
  note: string;
  updatedAt?: string;
  updatedBy?: string;
}

export interface HistoricExam {
  id: string;
  description: string;
  createdAt: string;
  files: FileInfoDTO[];
}
